src/main/kotlin/org/rust/ide/inspections/RsLocalInspectionTool.kt

```diff
@@ -106,10 +106,13 @@ class RsProblemsHolder(private val holder: ProblemsHolder) {
         rangeInElement: TextRange,
         vararg fixes: LocalQuickFix?
     ) {
-        if (element.existsAfterExpansion) {
+        if (element.existsAfterExpansion && checkHighlightType(highlightType)) {
             holder.registerProblem(element, message, highlightType, rangeInElement, *fixes)
         }
     }
+
+    private fun checkHighlightType(highlightType: ProblemHighlightType): Boolean =
```

Comment:
Maybe come up with a better name? right now it's not clear what it does

Reply:
Replaced with `showProblemWithType`

Reply:
Now, the name tells me that method should show some problem but actually, it doesn't show anything

I suggest something like `isAnnotationWithTypeAllowed`